ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_15/InterfaceIT.scala

```diff
@@ -13,36 +13,47 @@ import com.daml.ledger.api.testtool.infrastructure.Allocation.{
 import com.daml.ledger.api.testtool.infrastructure.Assertions._
 import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
 import com.daml.ledger.api.testtool.infrastructure.TransactionHelpers._
+import com.daml.ledger.api.v1.commands.Command.toJavaProto
 import com.daml.ledger.api.v1.value.{Identifier, Value}
-import com.daml.ledger.client.binding.Primitive
-import com.daml.ledger.test.semantic.Interface._
-import com.daml.ledger.test.semantic.{Interface1, Interface2, Interface3}
-import scalaz.Tag
+import com.daml.ledger.api.v1.commands.{Command => CommandV1}
+import com.daml.ledger.javaapi.data.Command
+import com.daml.ledger.javaapi.data.codegen.{ContractCompanion, Update}
+import com.daml.ledger.test.java.semantic.interface$._
+import com.daml.ledger.test.java.semantic.{interface1, interface2, interface3}
+
+import java.util.{List => JList}
+import scala.jdk.CollectionConverters._
 
 class InterfaceIT extends LedgerTestSuite {
+  implicit val tCompanion: ContractCompanion.WithKey[T.Contract, T.ContractId, T, String] =
+    T.COMPANION
 
-  private[this] val TId = Tag.unwrap(T.id)
-  private[this] val I1Id = Tag.unwrap(Interface1.I.id)
-  private[this] val I2Id = Tag.unwrap(Interface2.I.id)
-  private[this] val I3Id = Tag.unwrap(Interface3.I.id)
+  private[this] val TId = T.TEMPLATE_ID.toV1
+  private[this] val I1Id = interface1.I.TEMPLATE_ID.toV1
+  private[this] val I2Id = interface2.I.TEMPLATE_ID.toV1
+  private[this] val I3Id = interface3.I.TEMPLATE_ID.toV1
 
   // replace identifier with the wrong identifier for some of these tests
   private[this] def useWrongId[X](
-      command: Primitive.Update[X],
+      command: Update[X],
       id: Identifier,
-  ): Primitive.Update[X] = {
-    val exe = command.command.getExercise
+  ): JList[Command] = {
+    val commandV1 = CommandV1.fromJavaProto(command.commands.asScala.head.toProtoCommand)
+    val exe = commandV1.getExercise
     val arg = exe.getChoiceArgument.getRecord
-    command
-      .copy(
-        command = command.command.withExercise(
-          exe.copy(
-            templateId = Some(id),
-            choiceArgument = Some(Value.of(Value.Sum.Record(arg.copy(recordId = None)))),
+    JList.of(
+      Command
+        .fromProtoCommand(
+          toJavaProto(
+            commandV1.withExercise(
+              exe.copy(
```

Comment:
Here we too go from java -> scala, seemingly only to be able to access a useful scala function. In this example it's `copy`. How sbout staying in the java world for that?